docs/content/docs/HA.md

```diff
@@ -0,0 +1,55 @@
+---
+title: "HA Design considerations"
+description: "Open Service Mesh HA Design considerations"
+type: docs
+---
+# HA Design considerations
+
+OSM is built with High Availability and Fault Tolerance in mind. The following sections will thoroughly document how these are tackled in OSM.
+
+## High Availability and Fault Tolerance
+
+High Availability and Fault Tolerance are implemented and ensured by several design decisions and external mechanisms in OSM, which will be documented in the following points:
+
+### Statelessness
+OSM does not own or have any state-dependent data that needs to be saved at runtime; with the controlled exceptions of:
+- Root certificate: when no root certificate is provided and Tresor is used, OSM will ensure atomicity and idempotence when creating the root certificate through Kubernetes secret API, even when multiple instances are spawned.
+_Note: it is required and always assumed the root certificate is the same for all control plane components._
+- Envoy Bootstrap Certificates (used by the proxies to authenticate against the control plane): these are created and issued through Kubernetes Secrets API at injector webhook handling time, and are ensured idempotence when being created.
+
+Other than those exceptions, the rest of the configuration is built and fetched from Kubernetes.
+
+The domain state used to compute the traffic policies is entirely provided by the different providers (Kubernetes _et al_) and informers on the related objects OSM subscribes to.
+
+Multiple OSMs running will subscribe to the same objects and should generate the same view over the cluster as long as Kubernetes providers remain eventually consistent when updating through the informers, rendering mesh topology and policy enforcement eventually consistent at Kubernetes API pace.
+
+<p align="center">
+  <img src="/docs/content/docs/images/ha/ha1.png" width="400" height="350"/>
+</p>
+
+### Restartability: 
+The previous stateless design considerations should ensure OSM control plane is fully restartable.
+
+- A restarting instance will resynchronize all Kubernetes domain resources. Existing proxies will reconnect, and (assuming no changes occurred on the mesh topology or policy) the same configuration should be recomputed and pushed as a new version to the proxies.
+
+### Horizontal scaling
+OSM allows for horizontal scaling of its components (individually, for both the webhook/injector handling and the xDS control plane).
+- When OSM control plane is spawned with multiple replicaset, connecting proxies may be load-balanced and connected to any of the existing OSM instances running for the control plane.
+- Service certificates (used between proxies to TLS authenticate and communicate with each other) are short lived and kept only in runtime on control plane (though pushed as part of proxy xDS protocol when required).
```

Comment:
They are cached in memory. All pods participating in a Service will get the same service cert - same CN.  Cert creation is expensive - so we cache it.  If the controller restarted and we lost the Service cert cache, we'll create a new one. The new one will have the same CN, same Issuing CA, but a different serial number, which is not an issue. This of course is for Tresor. If we were to use cert-manager.io, Hashi Vault, or Azure Key Vault - this would be different.

Reply:
Certs are cached in memory for all certificate providers.

Reply:
> The new one will have the same CN, same Issuing CA, but a different serial number, which is not an issue
for the sake of discussion: It will eventually be an issue I believe. Restarting and re-issuing all certificates for all (connected) proxies and pushing all updates is going to take quite the toll at larger numbers.